changelog.txt

```diff
@@ -2,6 +2,7 @@
  * Fixed a bug making it impossible to read Realms created by iOS if a class contained primary keys (#1703).
  * Added Realm.isEmpty().
  * Fixed bug making it impossible to convert a field to being required during a migration.
+ * RealmQuery.isNull() and RealmQuery.isNotNull() now throw IllegalArgumentException instead of RealmError if the query target is a nested link field.
```

Comment:
@cmelchior How about `if the fieldname is a linked field and the last element is a link.`